x-pack/qa/repository-old-versions-compatibility/src/javaRestTest/java/org/elasticsearch/lucene/ArchiveIndexCompatibilityIT.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.lucene;
+
+import org.apache.http.util.EntityUtils;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.repositories.fs.FsRepository;
+import org.elasticsearch.test.cluster.util.Version;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+import java.util.stream.IntStream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static org.elasticsearch.test.rest.ObjectPath.createFromResponse;
+
+public class ArchiveIndexCompatibilityIT extends AbstractArchiveIndexCompatibilityTestCase {
+
+    static {
+        clusterConfig = config -> config.setting("xpack.license.self_generated.type", "trial");
+    }
+
+    public ArchiveIndexCompatibilityIT(Version version) {
+        super(version);
+    }
+
+    public void testArchiveIndicesV5() throws Exception {
```

Comment:
When this is run, we will already have upgraded, correct? Would it make sense to run basic tests after restore but before the upgrade?

I am also a bit anxious about sequencing here, because we may end up restoring directly after upgrade without noticing if we mess up the logic? 

Additionally, would it make sense to run a duel test, to say make sure that the index can be restored anew in 9, and keeps working when restored to 8, upgraded next to 9? We could compare functionality between these two.